File: BaekjoonAlgorithms/03-ForLoop/C/1924.c
/*2007년 x월 y일의 요일 구하기*/

#include <stdio.h>

int main()
{
	int x, y;
	scanf_s("%d %d", &x, &y);

	int cnt = 0;

	for (int i = 1; i <= x; i++) {
		switch (i) {
		case 1:
		case 3:
		case 5:
		case 7:
		case 8:
		case 10:
		case 12:
			for (int j = 1; j <= 31; j++) {
				cnt++;
				if (i == x && j == y) break;
			}
			break;
		case 4:
		case 6:
		case 9:
		case 11:
			for (int j = 1; j <= 30; j++) {
				cnt++;
				if (i == x && j == y) break;
			}
			break;
		case 2:
			for (int j = 1; j <= 28; j++) {
				cnt++;
				if (i == x && j == y) break;
			}
			break;
		}

	}
	cnt %= 7;
	switch (cnt) {
	case 1: printf("MON\n"); break;
	case 2: printf("TUE\n"); break;
	case 3: printf("WED\n"); break;
	case 4: printf("THU\n"); break;
	case 5: printf("FRI\n"); break;
	case 6: printf("SAT\n"); break;
	case 0: printf("SUN\n"); break;
	}
}

File: BaekjoonAlgorithms/03-ForLoop/python/1924.py
# 2007년 날짜에 따른 요일 구하기

x, y = map(int, input().split())  # 월, 일 입력
total_day = y  # 2007년 1월 1일로부터 지난 일수


def x_day(x):  # x월의 일수
    if x == 1 or x == 3 or x == 5 or x == 7 or x == 8 or x == 10 or x == 12:
        day = 31
    elif x == 4 or x == 6 or x == 9 or x == 11:
        day = 30
    elif x == 2:
        day = 28

    return day


def weekday(total_day):
    for i in range(1, x):
        total_day += x_day(i)

    if total_day % 7 == 1:
        weekday = 'MON'
    elif total_day % 7 == 2:
        weekday = 'TUE'
    elif total_day % 7 == 3:
        weekday = 'WED'
    elif total_day % 7 == 4:
        weekday = 'THU'
    elif total_day % 7 == 5:
        weekday = 'FRI'
    elif total_day % 7 == 6:
        weekday = 'SAT'
    else:
        weekday = 'SUN'
    
    return weekday


print(weekday(total_day))
